src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -52,69 +52,415 @@ public class RepoConfiguration {
+         *
+         * @param displayName Display name of the repository.
+         * @return This builder object.
+         */
+        public RepoBuilder displayName(String displayName) {
+            this.displayName = displayName;
+            return this;
+        }
+
+        /**
+         * Updates the {@code outputFolderName} for {@code RepoConfiguration}.
+         *
+         * @param outputFolderName Output folder name of the repository.
+         * @return This builder object.
+         */
+        public RepoBuilder outputFolderName(String outputFolderName) {
+            this.outputFolderName = outputFolderName;
+            return this;
+        }
+
+        /**
+         * Updates the {@code extraOutputFolderName} for {@code RepoConfiguration}.
+         *
+         * @param extraOutputFolderName Extra output folder name of the repository.
+         * @return This builder object.
+         */
+        public RepoBuilder extraOutputFolderName(String extraOutputFolderName) {
+            this.extraOutputFolderName = extraOutputFolderName;
+            return this;
+        }
+
+        /**
+         * Updates the {@code zoneId} for {@code RepoConfiguration}.
+         *
+         * @param zoneId Time-zone of the repository.
+         * @return This builder object.
+         */
+        public RepoBuilder zoneId(ZoneId zoneId) {
+            this.zoneId = zoneId;
+            return this;
+        }
+
+        /**
+         * Updates the {@code sinceDate} for {@code RepoConfiguration}.
+         *
+         * @param sinceDate Starting date of analysis.
+         * @return This builder object.
+         */
+        public RepoBuilder sinceDate(LocalDateTime sinceDate) {
+            this.sinceDate = sinceDate;
+            return this;
+        }
+
+        /**
+         * Updates the {@code untilDate} for {@code RepoConfiguration}.
+         *
+         * @param untilDate Ending date of analysis.
+         * @return This builder object.
+         */
+        public RepoBuilder untilDate(LocalDateTime untilDate) {
+            this.untilDate = untilDate;
+            return this;
+        }
+
+        /**
+         * Updates the {@code repoFolderName} for {@code RepoConfiguration}.
+         *
+         * @param repoFolderName Folder name of the repository.
+         * @return This builder object.
+         */
+        public RepoBuilder repoFolderName(String repoFolderName) {
+            this.repoFolderName = repoFolderName;
+            return this;
+        }
+
+        /**
+         * Updates the {@code fileTypeManager} for {@code RepoConfiguration}.
+         *
+         * @param fileTypeManager List of file types and groupings permitted.
+         * @return This builder object.
+         */
+        public RepoBuilder fileTypeManager(FileTypeManager fileTypeManager) {
+            this.fileTypeManager = fileTypeManager;
+            return this;
+        }
```

Comment:
Consider accepting a `List<FileType>` directly in the builder and internally creating the `FileTypeManager`, for simplicity and maintainability. This way, we encapsulate `FileTypeManager` creation and make the API easier for users, keeping the builder's usage clean and hiding unnecessary complexity.
```
public RepoBuilder formats(List<FileType> formats) {
    this.fileTypeManager = new FileTypeManager(formats);
    return this;
}
```